### PR TITLE
notification for appointments

### PR DESCRIPTION
**Pull request by mdah90**

### CODE REVIEW

File: functions/index.js [hunk 1]
```diff
@@ -1,174 +1,224 @@
...
+    return adminTokens;
+}
 
-    // Check if file is after resizer
-    if (!filePath.endsWith('_600x600.png')) {
-        // Updating only in case of resized image
-        return;
+// Update the admin when client creates new appointment and update the client when
+// admin creates new appointment for him
+async function HandleNewAppointment(snap, context) {
```
Comment (ghost) [inline functions/index.js]: functions starts with lower case letters e.g. handleNewAppointment :D

File: functions/index.js [hunk 2]
```diff
@@ -1,174 +1,224 @@
...
+    // Getting details of the new document
+    const appointmentId = context.params.appointmentId;
+    const newAppointmentData = snap.data();
+    const clientDocID = newAppointmentData.clientDocID;
+    if ( isClient(newAppointmentData.creator) ) {
+       console.log('Appointment created by client. need to notify the admin');
+       const clientName = newAppointmentData.clientName ?? 'User';
+       const title = clientName + ' Created new appointment';
+       const services = newAppointmentData.services ? newAppointmentData.services.length : 0;
+       const msg = clientName + ' created new appointment for ' +  services + ' services'
```
Comment (ghost) [inline functions/index.js:51]: Im not sure we want to show all services in the appointment in the notification body

Comment (ghost) [reply functions/index.js:51]: but lets keep it this way for now

File: functions/index.js (hunk 1 above)
Comment (mdah90) [reply functions/index.js]: indeed

File: functions/index.js (hunk 2 above)
Comment (mdah90) [reply functions/index.js:51]: This is just the number of services we need to decide what messages do we want .. i just put it to see that we get the correct data from appointment

File: functions/index.js [hunk 3]
```diff
@@ -1,174 +1,224 @@
...
+async function HandleAppointmentUpdate(change, context) {
+    // Get the updated object representing the updated document
+    const newValue = change.after.data();
+    const appointmentId = change.after.id;
+    // Get the previous object before this updated document
+    const previousValue = change.before.data();
+    // check the editor
+    const changedByClient = isClient(newValue.lastEditor);
+    const changedByBusiness = isBusiness(newValue.lastEditor);
+    const clientDocID = newValue.clientDocID;
```
Comment (ghost) [inline functions/index.js:102]: Currently, Admin can change client in update screen, hence, you will send update message to "new" client selected

I think we should strict the behavior here, client CAN NOT be changed changed, the expected flow should be:
1. cancel old appointment
2. create new one with new client

File: functions/index.js [hunk 4]
```diff
@@ -1,174 +1,224 @@
...
+                 sound: 'default'
+               },
+               data: {
+                 client_id: clientDocID,
+                 appointment_id: appointmentId,
+                 category: 'NotificationCategory.appointment',
+               }
+           };
+           const clientTokens = await getClientTokens(clientDocID);
+           if(clientTokens) {
```
Comment (ghost) [inline functions/index.js:176]: what if client is the one who cancelled the appointment ?
we should send the msg to admin

Comment (ghost) [reply functions/index.js:176]: sorry, forget this, Its declined status

**Review by ghost — APPROVED** [review]

**Pull request merged**